src/client/eppo-client.ts

```diff
@@ -82,168 +82,174 @@ export default class EppoClient implements IEppoClient {
   private queuedEvents: IAssignmentEvent[] = [];
   private assignmentLogger: IAssignmentLogger | undefined;
 
-  constructor(private configurationStore: IConfigurationStore) {}
+  constructor(private configurationStore: IConfigurationStore) { }
 
   public getAssignment(
     subjectKey: string,
-    experimentKey: string,
+    flagKey: string,
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     subjectAttributes: Record<string, any> = {},
     assignmentHooks?: IAssignmentHooks | undefined,
   ): string | null {
-    const assignment = this.getAssignmentInternal(
+    const { allocationKey, assignment } = this.getAssignmentInternal(
       subjectKey,
-      experimentKey,
+      flagKey,
       subjectAttributes,
       assignmentHooks,
       ValueType.StringType,
     );
-    assignmentHooks?.onPostAssignment(experimentKey, subjectKey, assignment);
+    assignmentHooks?.onPostAssignment(flagKey, subjectKey, assignment, allocationKey);
 
-    if (assignment !== null)
-      this.logAssignment(experimentKey, assignment, subjectKey, subjectAttributes);
+    if (assignment !== null && allocationKey !== null)
+      this.logAssignment(flagKey, allocationKey, assignment, subjectKey, subjectAttributes);
 
     return assignment?.stringValue ?? null;
   }
 
   public getStringAssignment(
     subjectKey: string,
-    experimentKey: string,
+    flagKey: string,
     subjectAttributes: Record<string, EppoValue> = {},
     assignmentHooks?: IAssignmentHooks | undefined,
   ): string | null {
-    const assignment = this.getAssignmentInternal(
+    const { allocationKey, assignment } = this.getAssignmentInternal(
       subjectKey,
-      experimentKey,
+      flagKey,
       subjectAttributes,
       assignmentHooks,
       ValueType.StringType,
     );
-    assignmentHooks?.onPostAssignment(experimentKey, subjectKey, assignment);
+    assignmentHooks?.onPostAssignment(flagKey, subjectKey, assignment, allocationKey);
 
-    if (assignment !== null)
-      this.logAssignment(experimentKey, assignment, subjectKey, subjectAttributes);
+    if (assignment !== null && allocationKey !== null)
+      this.logAssignment(flagKey, allocationKey, assignment, subjectKey, subjectAttributes);
 
     return assignment?.stringValue ?? null;
```

Comment:
Yeah I feel like that's basically what we had with the `get_assignment_variation` function in the dynamically typed languages. I also liked the early return logic to prevent logging events. I felt it was more readable than keeping track of nulls and performing actions based on whether something is null. I might try cleaning this up